xray/src/pcap.rs

```diff
@@ -0,0 +1,74 @@
+use std::net::SocketAddrV4;
+
+use neptun::noise::Tunn;
+use pcap::Capture;
+use pnet::packet::{ipv4::Ipv4Packet, sll2::SLL2Packet, udp::UdpPacket, Packet as _};
+
+use crate::{client::Client, utils::Packet, XRayResult};
+
+#[derive(Debug)]
+pub struct PcapPacket {
+    pub ts: u128,
+    pub src: SocketAddrV4,
+    pub dst: SocketAddrV4,
+    pub was_decrypted: bool,
+    pub send_index: u64,
+}
+
+pub fn process_pcap(pcap_path: &str, mut tunn: Tunn) -> XRayResult<Vec<PcapPacket>> {
+    let mut packets = Vec::new();
+
+    let mut capture = Capture::from_file(pcap_path)?;
+    let mut decrypt_buf = vec![0; 1024];
+
+    'top_level: while let Ok(packet) = capture.next_packet() {
```

Comment:
label is mostly for clarity?, as i don't see any other iteration

Reply:
Leftover from a previous version that had nested loops. Removed